front-end/src/pages/SettingsPage.tsx

```diff
@@ -0,0 +1,96 @@
+import axios from 'axios';
+import React, { useState } from 'react';
+import {
+    Card,
+    CardBody,
+    Button,
+    Form,
+    FormGroup,
+    Input,
+    Label,
+} from 'reactstrap';
+
+const SettingsPage = (loggedInUser: any) => {
+    // https://reactstrap.github.io/components/form/#app
+    // https://medium.com/better-programming/easily-create-a-form-with-react-hooks-1cab17e2be0d
+    // setting state: https://stackoverflow.com/questions/45850550/accessing-data-from-axios-get-request
+
+    const initialInputState = { displayName: "", githubUrl: "" };
+    const [eachEntry, setEachEntry] = useState(initialInputState);
+    const { displayName, githubUrl } = eachEntry;
+    const authorObject = {
+        _id: "",
+        _type: "author",
+        displayName: "",
+        url: "",
+        github: "",
+    };
+    const [adata, setAdata] = useState(authorObject);
+
+    const handleInputChange = (e: any) => {
+        setEachEntry({ ...eachEntry, [e.target.name]: e.target.value });
+    };
+
+    const handleSubmit = (e: any) => {
+        e.preventDefault();
+
+        axios.get(process.env.REACT_APP_API_URL + "/api/authors/").then(res => {
+            res.data.results.forEach((author: any) => {
+                // test by changing loggedInUser to a hardcoded string
+                if (author.displayName == loggedInUser.loggedInUser.username) {
```

Comment:
The authors display name and the username of the user are completely different right? But this wont be needed anymore with @Jmmxp and @ChrisChrisLoLo change with the new endpoint since you'll have the author object

Reply:
They're only the same right now as of the PR since it's set that way in signals.py.
It's a good point though, we need @ChrisChrisLoLo probably to change the registration form to also send github/url and make sure we properly set those in signals.py

But yes, we'll have the author object from the server soon, PR just went up